### PR TITLE
fix: use shortenAddress for mini portfolio addresses

### PR DESCRIPTION
**Pull request by natew**
## Description

Closing https://github.com/Uniswap/interface/pull/6701 in favor of this and a separate PR for barrel exports.

Uses shortenAddress to truncate for mini portfolio addresses.

Why have shortenAddessStrict + non-strict? Mostly don't want to change code across many areas in subtle ways, so keeping existing functionality.

_Linear ticket:_ https://linear.app/uniswap/issue/WEB-2071/truncate-wallet-addresses-in-mini-portfolio
_Slack thread:_
_Relevant docs:_

### Before

CleanShot 2023-06-05 at 10 47 38@2x

### After

CleanShot 2023-06-05 at 10 58 04@2x

### Reproducing the error

1. Send money from a wallet
2. Check the Activity pane in the wallet sidebar to see shortened address

We should test the various screens this affects as well, so all the files using shortenAddress / shortenAddressStrict. Happy to organize a quick bug bash if that is useful:

- [ ] src/components/AccountDrawer/AuthenticatedHeader.tsx
- [ ] src/components/AccountDrawer/MiniPortfolio/Activity/ActivityRow.tsx
- [ ] src/components/Web3Status/index.tsx
- [ ] src/components/claim/AddressClaimModal.tsx
- [ ] src/nft/components/collection/ActivityCells.tsx
- [ ] src/nft/components/details/AssetActivity.tsx
- [ ] src/nft/components/details/AssetPriceDetails.tsx
- [ ] src/nft/components/details/DetailsContainer.tsx
- [ ] src/nft/components/details/detailsV2/InfoChips.tsx
- [ ] src/nft/components/details/detailsV2/TableRowComponent.tsx

### QA (ie manual testing)

- [ ] N/A

### Automated testing

- [ ] Unit test
- [ ] Integration/E2E test

### CODE REVIEW

**Review by tinaszheng — APPROVED** [review]
👀 when do we want `strict` vs not strict - seems like everything should just be not strict? and if we are expecting a correct address (eg like from a backend response) and we dont get it, maybe we should log instead of throwing and error and maybe crashing the page

File: src/components/AccountDrawer/MiniPortfolio/Activity/ActivityRow.tsx [hunk 1]
```diff
@@ -52,7 +53,7 @@ export function ActivityRow({
         descriptor={
           <ActivityRowDescriptor color="textSecondary">
             {descriptor}
-            {ENSName ?? otherAccount}
+            {ENSName ?? shortenAddress(otherAccount ?? '')}
```
Comment (cbachmeier) [inline src/components/AccountDrawer/MiniPortfolio/Activity/ActivityRow.tsx:56]: non-blocking nit: `shortenAddress` could accept an undefined address and return the empty string

File: src/utils/addresses.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    // Alphabetical letters must be made lowercase for getAddress to work.
+    // See documentation here: https://docs.ethers.io/v5/api/utils/address/
+    return getAddress(value.toLowerCase())
+  } catch {
+    return false
+  }
+}
+
+// Shortens an Ethereum address
+export function shortenAddress(address: string): string {
```
Comment (cbachmeier) [inline src/utils/addresses.ts:15]: was it intentional to remove the ability to set the number of chars at the beginning/end? I noticed `src/nft/components/details/DetailsContainer.tsx` for example used to only have two characters in the beginning but would receive 4 now.

File: src/utils/addresses.test.ts [hunk 3]
```diff
@@ -23,9 +23,27 @@ describe('utils', () => {
...
+    it('truncates middle characters even without prefix', () => {
+      expect(shortenAddressStrict('f164fc0ec4e93095b804a4795bbe1e041497b92a')).toBe('0xf164...b92a')
+    })
+
+    it('renders checksummed address', () => {
+      expect(shortenAddressStrict('0x2E1b342132A67Ea578e4E3B814bae2107dc254CC'.toLowerCase())).toBe('0x2E1b...54CC')
+    })
+  })
+
+  describe('#shortenAddress', () => {
```
Comment (cbachmeier) [inline src/utils/addresses.test.ts:44]: add a test for a for an alternate number of chars at the beginning and another for alternate at the end

File: src/utils/index.ts [hunk 4]
```diff
@@ -1,47 +1,3 @@
...
-    throw Error(`Invalid 'address' parameter '${address}'.`)
-  }
-
-  return new Contract(address, ABI, getProviderOrSigner(provider, account) as any)
-}
-
-export function escapeRegExp(string: string): string {
-  return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&') // $& means the whole matched string
-}
+export * from './addresses'
```
Comment (cbachmeier) [inline src/utils/index.ts:1]: chefs-kiss-emoji-2

**Review by cbachmeier — CHANGES_REQUESTED** [review]

File: src/utils/addresses.ts (hunk 2 above)
Comment (cbachmeier) [reply src/utils/addresses.ts:15]: This PR vs Staging
Screen Shot 2023-06-15 at 14 26 57
Screen Shot 2023-06-15 at 14 27 12

File: src/utils/addresses.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+/**
+ * Shorten an address and add 0x to the start if missing
+ * @param target
+ * @param charsStart amount of character to shorten (from both ends / in the beginning)
+ * @param charsEnd amount of characters to shorten in the end
+ * @returns formatted string
+ */
+function ellipseAddressAdd0x(address: string, charsStart = 4, charsEnd = 4): string {
```
Comment (JFrankfurt) [inline src/utils/addresses.ts:37]:
```suggestion
 * @param targetAddress
 * @param charsStart amount of character to shorten (from both ends / in the beginning)
 * @param charsEnd amount of characters to shorten in the end
 * @returns formatted string
 */
function ellipseAddressAdd0x(targetAddress: string, charsStart = 4, charsEnd = 4): string {
```

File: src/utils/addresses.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+/**
+ * Shorten a string with "..." in the middle
+ * @param target
+ * @param charsStart amount of character to shorten (from both ends / in the beginning)
+ * @param charsEnd amount of characters to shorten in the end
+ * @returns formatted string
+ */
+function ellipseMiddle(target: string, charsStart = 4, charsEnd = 4): string {
```
Comment (JFrankfurt) [inline src/utils/addresses.ts:50]:
```suggestion
 * @param targetAddress
 * @param charsStart amount of character to shorten (from both ends / in the beginning)
 * @param charsEnd amount of characters to shorten in the end
 * @returns formatted string
 */
function ellipseMiddle(targetAddress: string, charsStart = 4, charsEnd = 4): string {
```

File: src/utils/addresses.ts (hunk 5 above)
Comment (JFrankfurt) [reply src/utils/addresses.ts:37]: trying to unify the variable name with the docs, as you had in the implementation below

**Review by JFrankfurt — APPROVED** [review]

File: src/utils/addresses.ts (hunk 2 above)
Comment (natew) [reply src/utils/addresses.ts]: good catch!

**Comment by natew** [comment]
@cbachmeier thanks for review, addressed feedback

File: src/nft/components/collection/ActivityCells.tsx [hunk 7]
```diff
@@ -165,7 +165,7 @@ export const AddressCell = ({ address, desktopLBreakpoint, chainId }: AddressCel
         href={getExplorerLink(chainId ?? ChainId.MAINNET, address ?? '', ExplorerDataType.ADDRESS)}
         style={{ textDecoration: 'none' }}
       >
-        <Box onClick={(e) => e.stopPropagation()}>{address ? shortenAddress(address, 2, 4) : '-'}</Box>
+        <Box onClick={(e) => e.stopPropagation()}>{address ? shortenAddress(address) : '-'}</Box>
```
Comment (cbachmeier) [inline src/nft/components/collection/ActivityCells.tsx:168]: 2 chars at start here

File: src/nft/components/details/AssetActivity.tsx [hunk 8]
```diff
@@ -177,15 +177,15 @@ const AssetActivity = ({ events }: { events?: ActivityEvent[] }) => {
               <TD>
                 {fromAddress && (
                   <Link href={`https://etherscan.io/address/${fromAddress}`} target="_blank" rel="noopener noreferrer">
-                    {shortenAddress(fromAddress, 2, 4)}
+                    {shortenAddress(fromAddress)}
```
Comment (cbachmeier) [inline src/nft/components/details/AssetActivity.tsx:180]: 2 chars at start here

File: src/nft/components/details/AssetActivity.tsx [hunk 9]
```diff
@@ -177,15 +177,15 @@ const AssetActivity = ({ events }: { events?: ActivityEvent[] }) => {
...
                   </Link>
                 )}
               </TD>
 
               <TD>
                 {toAddress && (
                   <Link href={`https://etherscan.io/address/${toAddress}`} target="_blank" rel="noopener noreferrer">
-                    {shortenAddress(toAddress, 2, 4)}
+                    {shortenAddress(toAddress)}
```
Comment (cbachmeier) [inline src/nft/components/details/AssetActivity.tsx:188]: 2 chars at start here

File: src/nft/components/details/AssetDetails.tsx [hunk 10]
```diff
@@ -427,7 +427,7 @@ export const AssetDetails = ({ asset, collection }: AssetDetailsProps) => {
               target="_blank"
               rel="noopener noreferrer"
             >
-              {shortenAddress(asset.creator.address, 2, 4)}
+              {shortenAddress(asset.creator.address)}
```
Comment (cbachmeier) [inline src/nft/components/details/AssetDetails.tsx:430]: 2 chars at start here

File: src/nft/components/details/AssetPriceDetails.tsx [hunk 11]
```diff
@@ -425,7 +425,7 @@ export const AssetPriceDetails = ({ asset, collection }: AssetPriceDetailsProps)
             {asset.tokenType === 'ERC1155' ? (
               ''
             ) : (
-              <span> {isOwner ? 'You' : asset.ownerAddress && shortenAddress(asset.ownerAddress, 2, 4)}</span>
+              <span> {isOwner ? 'You' : asset.ownerAddress && shortenAddress(asset.ownerAddress)}</span>
```
Comment (cbachmeier) [inline src/nft/components/details/AssetPriceDetails.tsx:428]: 2 chars at start here

File: src/nft/components/details/DetailsContainer.tsx [hunk 12]
```diff
@@ -80,7 +80,7 @@ const DetailsContainer = ({ asset, collection }: { asset: GenieAsset; collection
         header="Contract address"
         body={
           <Center onClick={copy}>
-            {shortenAddress(address, 2, 4)} <CopyIcon size={13} />
+            {shortenAddress(address)} <CopyIcon size={13} />
```
Comment (cbachmeier) [inline src/nft/components/details/DetailsContainer.tsx:83]: 2 chars at start here

**Review by cbachmeier — CHANGES_REQUESTED** [review]

**Review by cbachmeier — APPROVED** [review]

**Pull request merged**